### PR TITLE
SUP-115-Create-Stream-Application-Audit

### PR DESCRIPTION
**Pull request by edgargarciamacrometa**

### CODE REVIEW

**Comment by edgargarciamacrometa** [comment]
I added collection.type='DOC', replication.type='GLOBAL' in some parts of the code, we do not support local collections at the moment, but CEP is taking local replication type as default, so the parameter needs to be defined in order to work. Also fixed this error integer -> int, 
Added example with CREATE TABLE GLOBAL SweetTable(product_id string, quantity int);
I think we don't support the following syntax: @sink(type="c8streams", stream="<STRING>", replication.type="<STRING>", @map(...)))

File: docs/cep/tutorials/create-stream-app.md [hunk 1]
```diff
@@ -75,13 +75,13 @@ To create a stream application follow the steps below:
 ### C8Streams
```
Comment (diana-macrometa) [inline docs/cep/tutorials/create-stream-app.md]:
```suggestion
### Create Streams
```

File: docs/cep/tutorials/create-stream-app.md [hunk 2]
```diff
@@ -117,12 +117,14 @@ Query Parameters:
 ### C8DB
```
Comment (diana-macrometa) [inline docs/cep/tutorials/create-stream-app.md]: It seems like this is the wrong heading. Maybe something about stream workers?

File: docs/cep/tutorials/create-stream-app.md [hunk 3]
```diff
@@ -117,12 +117,14 @@ Query Parameters:
...
+```
 
 Example:
-
-	CREATE SOURCE SweetProductionStream WITH (type="database", map.type='json') (name string, amount double);
+```sql
+   CREATE SOURCE SweetProductionStream WITH (type = 'database', collection='SweetProductionData', collection.type='DOC', replication.type='GLOBAL',  map.type='json') (name string, amount double);
+```
 
 Query Parameters:
```
Comment (diana-macrometa) [inline docs/cep/tutorials/create-stream-app.md]:
```suggestion
Query parameters:
```

File: docs/cep/tutorials/create-stream-app.md [hunk 4]
```diff
@@ -209,12 +211,17 @@ Query Parameters:
 ### C8DB
```
Comment (diana-macrometa) [inline docs/cep/tutorials/create-stream-app.md]: Avoid stacked headings, need an intro paragraph after the Table heading above. Also, I don't think we're using C8DB anymore, so this seems like the wrong heading text for this section.

File: docs/cep/tutorials/create-stream-app.md [hunk 5]
```diff
@@ -209,12 +211,17 @@ Query Parameters:
...
-	CREATE STORE SweetProductionCollection WITH (type="database", collection="SweetProductionCollection", replication.type="local", map.type='json') (strings);
+```sql
+   CREATE STORE SweetProductionCollection WITH (type="database", collection="SweetProductionCollection", replication.type="local", collection.type="DOC", map.type='json') (name string, amount double);
+```
+Or equivalent using TABLE:
+```sql
+   CREATE TABLE GLOBAL SweetProductionCollection(name string, amount double);
+```
 
 Stream applications will use the c8db with the default query parameters explained in the chart below.
```
Comment (diana-macrometa) [inline docs/cep/tutorials/create-stream-app.md]: Two things:
1. In all cases, replace "stream applications" with "stream workers." They used to call them stream applications, they don't anymore.
2. "C8DB" is also outdated terminology. Replace with "Macrometa" or the exact name of what is being accessed, like a collection.

File: docs/cep/tutorials/create-stream-app.md [hunk 6]
```diff
@@ -238,10 +245,10 @@ Stream applications will use the c8db with the default query parameters explaine
 </tr>
 <tr class="even">
 <td>replication.type</td>
-<td>Specifies if the replication type of the c8db collection. Possible values can be `local` and `global`</td>
+<td>Specifies if the replication type of the c8db collection. At the moment local collections are not allowed, type must be `global`</td>
```
Comment (diana-macrometa) [inline docs/cep/tutorials/create-stream-app.md]:
```suggestion
<td>Specifies if the replication type of the collection. <b>Note:</b> Type must be `global`. Local collections are not currently allowed.</td>
```

**Review by diana-macrometa — CHANGES_REQUESTED** [review]

**Comment by diana-macrometa** [comment]
When this is ready for re-review, request one by clicking the circling arrows next to my name.

File: docs/cep/tutorials/create-stream-app.md [hunk 7]
```diff
@@ -38,51 +38,52 @@ Stream applications are declarative specs that define the processing logic to pr
 <ul>
 <li>Stateless query: Queries that only consider currently incoming events when generating an output. e.g., filters</li>
 <li>Stateful query: Queries that consider both currently incoming events as well as past events when generating an output. e.g., windows, sequences, patterns, etc.</li>
-<li>Partitions: Collections of stream definitions and queries separated from each other within a Stream application for the purpose of processing events in parallel and in isolation</li>
+<li>Partitions: Collections of stream definitions and queries separated from each other within a Stream worker for the purpose of processing events in parallel and in isolation</li>
```
Comment (diana-macrometa) [inline docs/cep/tutorials/create-stream-app.md]:
```suggestion
<li>Partitions: Collections of stream definitions and queries separated from each other within a stream worker for the purpose of processing events in parallel and in isolation</li>
```

File: docs/cep/tutorials/create-stream-app.md [hunk 8]
```diff
@@ -38,51 +38,52 @@ Stream applications are declarative specs that define the processing logic to pr
...
 
-1. Open the GUI. Click the **Stream Apps** tab.
-1. Click **New** to define a new stream application.
-1. Type a **Name** for the stream application. For example, `SweetProductionAnalysis`.
+1. Open the GUI. Click the **Stream Workers** tab.
+1. Click **New Stream Worker** to define a new stream worker.
+1. Type a **Name** for the stream worker. For example, `SweetProductionAnalysis`.
 1. Type a **Description**.
-1. Add the following sample stream application.
+1. Add the following sample stream worker.
```
Comment (diana-macrometa) [inline docs/cep/tutorials/create-stream-app.md]:
```suggestion
1. Add the following sample stream worker:
```

File: docs/cep/tutorials/create-stream-app.md [hunk 9]
```diff
@@ -38,51 +38,52 @@ Stream applications are declarative specs that define the processing logic to pr
...
+	
 	INSERT INTO ProductionAlertStream
 	SELECT *
 	FROM SweetProductionStream;
 	```
 
-1. Click `Save` to save the stream app.
+1. Click `Save` to save the stream worker.
 1. Select all the regions to deploy your application in.
 1. Click on `Save`.
```
Comment (diana-macrometa) [inline docs/cep/tutorials/create-stream-app.md]:
```suggestion
1. Click **Save**.
```

File: docs/cep/tutorials/create-stream-app.md [hunk 10]
```diff
@@ -38,51 +38,52 @@ Stream applications are declarative specs that define the processing logic to pr
...
 	CREATE SINK ProductionAlertStream WITH (type= 'stream', stream='ProductionAlertStream', map.type='json') (name string, amount double);
-
+	
 	INSERT INTO ProductionAlertStream
 	SELECT *
 	FROM SweetProductionStream;
 	```
 
-1. Click `Save` to save the stream app.
+1. Click `Save` to save the stream worker.
```
Comment (diana-macrometa) [inline docs/cep/tutorials/create-stream-app.md]:
```suggestion
1. Click **Save** to save the stream worker.
```

File: docs/cep/tutorials/create-stream-app.md [hunk 11]
```diff
@@ -38,51 +38,52 @@ Stream applications are declarative specs that define the processing logic to pr
...
 1. Select all the regions to deploy your application in.
```
Comment (diana-macrometa) [inline docs/cep/tutorials/create-stream-app.md:70]: Is there a minimum of two here? I know there is a minimum for fabrics.

File: docs/cep/tutorials/create-stream-app.md [hunk 12]
```diff
@@ -38,51 +38,52 @@ Stream applications are declarative specs that define the processing logic to pr
...
 1. Click on `Save`.
 
 ## Source
 
-### C8Streams
+For this example, a source can be of type stream or database. The incoming data is used for processing. A source-of-type database is best if you need to store incoming data. Otherwise, you can use a source-type stream.
+### Create Streams
```
Comment (diana-macrometa) [inline docs/cep/tutorials/create-stream-app.md:77]:
```suggestion

### Create Streams
```

File: docs/cep/tutorials/create-stream-app.md [hunk 13]
```diff
@@ -38,51 +38,52 @@ Stream applications are declarative specs that define the processing logic to pr
...
 
 Syntax:
-
-	CREATE SOURCE SourceName WITH (type="stream", stream.list="STRING", replication.type="STRING", map.type='type') (strings);
-
+```sql
+   CREATE SOURCE SourceName WITH (type="stream", stream.list="STRING", replication.type="STRING", map.type='type') (strings);
+```
 Example:
```
Comment (diana-macrometa) [inline docs/cep/tutorials/create-stream-app.md:85]:
```suggestion

Example:
```

File: docs/cep/tutorials/create-stream-app.md [hunk 14]
```diff
@@ -38,51 +38,52 @@ Stream applications are declarative specs that define the processing logic to pr
...
 
-### C8Streams
+For this example, a source can be of type stream or database. The incoming data is used for processing. A source-of-type database is best if you need to store incoming data. Otherwise, you can use a source-type stream.
+### Create Streams
 
 Syntax:
-
-	CREATE SOURCE SourceName WITH (type="stream", stream.list="STRING", replication.type="STRING", map.type='type') (strings);
-
+```sql
```
Comment (diana-macrometa) [inline docs/cep/tutorials/create-stream-app.md:81]:
```suggestion

```sql
```

File: docs/cep/tutorials/create-stream-app.md [hunk 15]
```diff
@@ -114,17 +115,21 @@ Query Parameters:
...
-Syntax:
+You can create collections with your stream worker, and store incoming data in it for further processing, the syntax to achieve that is shown below:
 
-	CREATE SOURCE SourceName WITH (type="database", collection="STRING", replication.type="STRING", collection.type="STRING", map.type='type') (strings);
+Syntax:
+```sql
+   CREATE SOURCE SourceName WITH (type="database", collection="STRING", replication.type="STRING", collection.type="STRING", map.type='type') (strings);
+```
 
 Example:
```
Comment (diana-macrometa) [inline docs/cep/tutorials/create-stream-app.md:132]:
```suggestion

Example:
```

File: docs/cep/tutorials/create-stream-app.md [hunk 16]
```diff
@@ -114,17 +115,21 @@ Query Parameters:
...
 </table>
 
-### C8DB
+### Create collections
 
-Syntax:
+You can create collections with your stream worker, and store incoming data in it for further processing, the syntax to achieve that is shown below:
 
-	CREATE SOURCE SourceName WITH (type="database", collection="STRING", replication.type="STRING", collection.type="STRING", map.type='type') (strings);
+Syntax:
```
Comment (diana-macrometa) [inline docs/cep/tutorials/create-stream-app.md:125]:
```suggestion
Syntax:

```

File: docs/cep/tutorials/create-stream-app.md [hunk 17]
```diff
@@ -206,17 +213,24 @@ Query Parameters:
...
+```
 
-	CREATE STORE SweetProductionCollection WITH (type="database", collection="SweetProductionCollection", replication.type="local", map.type='json') (strings);
+Or equivalent using STORE:
+```sql
+   CREATE STORE SweetProductionCollection WITH (type="database", collection="SweetProductionCollection", replication.type="local", collection.type="DOC", map.type='json') (name string, amount double);
+```
 
-Stream applications will use the c8db with the default query parameters explained in the chart below.
+Stream worker will use the macrometa collections with the default query parameters explained in the chart below.
```
Comment (diana-macrometa) [inline docs/cep/tutorials/create-stream-app.md]:
```suggestion
The stream worker will use the Macrometa collections with the default query parameters explained in the chart below.
```

**Review by diana-macrometa — CHANGES_REQUESTED** [review]
A few more minor things, mostly formatting. We're getting close!

File: docs/cep/tutorials/create-stream-app.md (hunk 11 above)
Comment (edgargarciamacrometa) [reply docs/cep/tutorials/create-stream-app.md:70]: A client can have just 2 in playground active, but many more stream workers not enabled

**Review by diana-macrometa — APPROVED** [review]
Looks good to me

**Pull request merged**